onboard/catkin_ws/src/controls/include/thruster_allocator.h

```diff
@@ -0,0 +1,92 @@
+#ifndef THRUSTER_ALLOCATOR_H
+#define THRUSTER_ALLOCATOR_H
+
+#include <OsqpEigen/OsqpEigen.h>
+
+#include <Eigen/Dense>
+#include <string>
+
+class ThrusterAllocator {
+   private:
+    /**
+     * @brief Maximum absolute value allocation allowed for any thruster.
+     */
+    double max_alloc = 1.0;
+
+    /**
+     * @brief Wrench matrix.
+     *
+     * @details 6 x n matrix where n is the number of thrusters.
+     * Each column represents the force and torque generated by a single thruster for each axis.
+     * Each row represents an axis, in the order of x, y, z, roll, pitch, yaw from top to bottom.
+     */
+    Eigen::MatrixXd wrench;
+
+    /**
+     * @brief Pseudoinverse of wrench matrix.
+     *
+     * @details n x 6 matrix where n is the number of thrusters.
+     */
+    Eigen::MatrixXd wrench_pinv;
+
+    /**
+     * @brief OSQP solver.
+     *
+     * @details Solver for quadratic programming problems.
+     */
+    OsqpEigen::Solver solver;
+
+    /**
+     * @brief Get unconstrained thruster allocations for given set power by returning
+     *  `wrench_pinv * set_power`.
+     *
+     * @param set_power Set power.
+     * @param unconstrained_allocs Unconstrained thruster allocations.
+     */
+    void get_pseudoinverse_solution(const Eigen::VectorXd &set_power, Eigen::VectorXd &unconstrained_allocs);
```

Comment:
Since `get_pseudoinverse_solution` and `get_qp_solution` are only used in one context, we can rename them `get_unconstrained_allocs` and `get_constrained_allocs` to make it clear what they're being used for.

Reply:
The pseudoinverse solution is sometimes used for the unconstrained and constrained allocs, and the qp solution is not always used for the constrained allocs. So, I think it's better to name them based on the method of computation. One can look at the `allocate_thrusters` function to see what they're being used for.